test/test_before_test.py

```diff
@@ -0,0 +1,56 @@
+from .template_projects.c import CTemplateProject
+from . import utils
+
+before_test_project = CTemplateProject()
+before_test_project.files['test/spam_test.py'] = r'''
```

Comment:
I was trying to avoid the indent and `textwrap.dedent()`. Also, I think it's nice to expose the workings of TemplateProject to the tests, so they can fiddle with things at runtime, as seen in the cpp_standards test